pysat/_instrument.py

```diff
@@ -2050,6 +2141,16 @@ def to_netcdf4(self, fname=None, base_instrument=None, epoch_name='Epoch',
         if hasattr(self._export_meta_post_processing, '__call__'):
             export_meta = self._export_meta_post_processing(export_meta)
 
+        # check if there are multiple variables with same characters
+        # but with different case
+        variables = self.variables
+        variables = [var.lower() for var in variables]
```

Comment:
Why not combine these two: `variables = [var.lower() for var in self.variables]`?  Alternatively: `self.variables = [var.lower() for var in self.variables]` and using `self.variables` elsewhere.

Reply:
self.variables is a property that returns the names as in the underlying pandas or xarray object. I've made the code more efficient, as suggested.